src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -273,6 +349,62 @@ public class PullRequestHandler : IHttpHandler
         return success;
     }
 
+    // Read configuration
+    // Precedence is: 
+    // 1. Environment variable
+    // 2. AppSettings in web.config
+    // 
+    private bool ReadConfiguration(StringBuilder result)
+    {
+        if (_configurationRead)
+        {
+            // No locking needed, reading this in twice, even on multiple concurrent threads, doesn't do harm
+            return true;
+        }
+
+        NameValueCollection appSettings = ConfigurationManager.AppSettings;
+
+        string gitExecutablePath = Environment.GetEnvironmentVariable(_gitExecutablePathConfigurationKeyString);
```

Comment:
More code more chances for failure. I really want to cut PRService code to bare minimum :)

Will reading environment variable cause more permission to be needed by the app pool?